src/lib/components/Swap/SwapButton.tsx

```diff
@@ -129,18 +131,46 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
 
   //@TODO(ianlapham): add a loading state, process errors
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
+
   const onConfirm = useCallback(() => {
     swapCallback?.()
-      .then((transactionResponse) => {
-        // TODO(ianlapham): Add the swap tx to transactionsAtom
-        console.log(transactionResponse)
-        setDisplayTxHash(transactionResponse.hash)
+      .then((response) => {
+        setDisplayTxHash(response.hash)
+        if (inputCurrency && outputCurrency && inputCurrencyAmount && outputCurrencyAmount) {
```

Comment:
You never actually use 'inputCurrency' and 'outputCurrency' here, you should omit them and not destructure 'outputCurrency'.